os/pm/pm_idle.c

```diff
@@ -65,12 +65,29 @@ void pm_idle(void)
 	enum pm_state_e newstate;
 #ifdef CONFIG_PM_TIMEDWAKEUP
 	clock_t delay;
+#endif
+#ifdef CONFIG_SMP
+	int cpu;
+	FAR struct tcb_s *tcb = NULL;
 #endif
 	/* Decide, which power saving level can be obtained */
 	flags = enter_critical_section();
 	newstate = pm_checkstate();
 	/* Perform state-dependent logic here */
-	pmvdbg("newstate= %d\n", newstate);
+	/* For SMP case, we need to check secondary core status
+	   If secondary core status is not in idle thread, abort
+	   the sleep and check again on next cycle
+	*/
+#ifdef CONFIG_SMP
+	if (newstate == PM_SLEEP) {
+		for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
+			tcb = current_task(cpu);
```

Comment:
Even if we check the cpu 1 idle state here, there are many steps happening, before SGI4 is actually sent to cpu1. Due to this time difference, it is possible that cpu1 might start running a different task. So, I think we should check for the cpu1 idle state in the SGI4 handler only.

Reply:
Hi @kishore-sn , as I have discussed with @ewoodev , we think that pm_idle could check the state of cores other than primary core (ie. only primary core conduct PM logic). Thus, we will utilize SGI2 to control other core status, to prevent from context switch or other interrupt

Reply:
@edwakuwaku I understand Kishore's concern. Since we have paused the CPU, but it may already be in an tick IRQ or late to be paused, the checked CPU state may differ.

cpu sched checking in the SGI4 is better.
Thank you for consider and explain.

How about requesting SGI4 in up_cpu_hotplug and returning it with a flag? 
If the up_cpu_hotplug returns fail, we don't need to call pm_changestate.